server/src/server.ts

```diff
@@ -108,29 +110,90 @@ connection.onInitialized((_params) => {
     connection.workspace.getConfiguration('yseopml.parseAllProjectFilesAtStartup').then((_confValue) => {
         parseAllProjectFilesAtStartup = _confValue;
         if (!parseAllProjectFilesAtStartup) {
-            connection.console.log('Not parsing project files at startup.');
             return;
         }
-        connection.console.log('Parsing project files at startup.');
-        connection.workspace.getWorkspaceFolders().then((_value) => {
-            const workspacePath = url.fileURLToPath(_value[0].uri);
-            glob('**/project.kao', {
-                cwd: workspacePath,
+        // The value of yseopml.kaoFiles is useful here
+        // only when we want to parse the whole project.
+        connection.workspace
+            .getConfiguration('yseopml.kaoFiles')
+            .then((_kaoFiles) => {
+                // initialize array kaoFilesToParse
+                kaoFilesToParse = _kaoFiles;
             })
-                .then((_matches) => {
-                    for (const filePath of _matches) {
-                        if (openProjectFile(workspacePath, path.join(workspacePath, filePath))) {
-                            // There should be only one `project.kao` file. If we found a good candidate, stop the loop.
-                            return;
-                        }
-                    }
-                })
-                .catch((_error) => console.error(_error));
-            parseAutoExportedFiles(workspacePath);
-        });
+            .then(() => {
+                parseWholeProject();
+            });
     });
 });
 
+/**
+ * Parse the whole project YML files using kao files as starting point.
+ *
+ * @note This function is used only at the extension's initialization,
+ * but can also be used when updating some settings in the future.
+ */
+function parseWholeProject(): void {
+    if (!parseAllProjectFilesAtStartup) {
+        connection.console.log('Not parsing whole project files.');
```

Comment:
maybe useless sugg because maybe useless `if` anyway

```suggestion
        connection.console.log('Not parsing every YML file of the project.');
```